Vector{Pair} - display errors for element type Pair{Any,Vector}

```
julia> pairs = Vector{Pair}()
0-element Array{Pair,1}

julia> push!(pairs, ""=>[1])
1-element Array{Pair,1}:
Error showing value of type Array{Pair,1}:
ERROR: AssertionError: Array{Int64,1} is not a subtype of Pair
Stacktrace:
 [1] typeinfo_prefix(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Array{Int64,1}) at ./arrayshow.jl:480
 [2] show_vector(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Array{Int64,1}, ::Char, ::Char) at ./arrayshow.jl:433 (repeats 2 times)
 [3] show at ./arrayshow.jl:447 [inlined]
 [4] show(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Pair{String,Array{Int64,1}}) at ./show.jl:424

julia> pairs[1]
"" => [1]
```

--- a/base/show.jl
+++ b/base/show.jl
@@ -577,12 +577,16 @@ function show(io::IO, p::Pair)
     iocompact = IOContext(io, :compact => get(io, :compact, true))
     has_tight_type(p) || return show_default(iocompact, p)
 
+    typeinfo = get(io, :typeinfo, Any)
+    typeinfos = typeinfo <: Pair && p isa typeinfo ?
+        (fieldtype(typeinfo, 1), fieldtype(typeinfo, 2)) : (Any, Any)
+
     isdelimited(iocompact, p.first) || print(io, "(")
-    show(iocompact, p.first)
+    show(IOContext(iocompact, :typeinfo => typeinfos[1]), p.first)
     isdelimited(iocompact, p.first) || print(io, ")")
     print(io, compact ? "=>" : " => ")
     isdelimited(iocompact, p.second) || print(io, "(")
-    show(iocompact, p.second)
+    show(IOContext(iocompact, :typeinfo => typeinfos[2]), p.second)
     isdelimited(iocompact, p.second) || print(io, ")")
     nothing
 end

--- a/test/show.jl
+++ b/test/show.jl
@@ -1158,6 +1158,9 @@ end
 
     # issue #25857
     @test repr([(1,),(1,2),(1,2,3)]) == "Tuple{$Int,Vararg{$Int,N} where N}[(1,), (1, 2), (1, 2, 3)]"
+
+    # issues #25466 & #26256
+    @test replstr([:A => [1]]) == "1-element Array{Pair{Symbol,Array{$Int,1}},1}:\n :A => [1]"
 end
 
 @testset "#14684: `display` should print associative types in full" begin

